### PR TITLE
added boilerplate code

### PR DESCRIPTION
**Pull request by Bashar200**

### CODE REVIEW

File: src/agendaInit.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        (event_name = _.attrs.name),
+        (body = _.attrs.data.body),
+        (call_back_url = _.attrs.data.call_back_url),
+      );
+    });
+  });
+  await agenda.start();
+});
+
+const graceful = () => {
```
Comment (mehul-mittal-credgenics) [inline src/agendaInit.js:29]: graceful shutdown

File: src/app.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+const agenda = require('./agendaInit');
+const routes = require('./routes/route');
+const ApiError = require('./utils/ApiError');
+const { BASE_ROUTE, PORT, MONGO_URI } = require('./settings');
+const logger = require('./logger');
+const { errorConverter, errorHandler } = require('./error');
+
+const app = express();
+
+app.use(express.json());
```
Comment (mehul-mittal-credgenics) [inline src/app.js:17]: understand the use cases of middleware before adding

File: src/error.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+const mongoose = require('mongoose');
+const httpStatus = require('http-status');
+const logger = require('./logger');
+const ApiError = require('./utils/ApiError');
+const { ENV } = require('./settings');
+
+const errorConverter = (err, req, res, next) => {
+  let error = err;
+  if (!(error instanceof ApiError)) {
```
Comment (mehul-mittal-credgenics) [inline src/error.js:9]: handle negative case

File: src/error.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      : httpStatus.INTERNAL_SERVER_ERROR;
+    const message = error.message || httpStatus[statusCode];
+    error = new ApiError(statusCode, message, false, err.stack);
+  }
+  next(error);
+};
+
+const errorHandler = (err, req, res, next) => {
+  let { statusCode, message } = err;
+  if (ENV === 'prod' && !err.isOperational) {
```
Comment (mehul-mittal-credgenics) [inline src/error.js:21]: requirement for env based checks ??

File: src/jobs/jobs_list/triggerEvent.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+const axios = require('axios');
```
Comment (mehul-mittal-credgenics) [inline src/jobs/jobs_list/triggerEvent.js:1]: nomenclature for files

File: src/jobs/jobs_list/triggerEvent.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+const logger = require('../../logger');
+
+module.exports = (
+  agenda,
+  url = '',
+  headers = {},
+  event_name = '',
+  body = {},
+  call_back_url = '',
+) => {
```
Comment (mehul-mittal-credgenics) [inline src/jobs/jobs_list/triggerEvent.js:11]: don't write callback with function

File: src/jobs/jobs_list/triggerEvent.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  agenda.define(event_name, async (job) => {
```
Comment (mehul-mittal-credgenics) [inline src/jobs/jobs_list/triggerEvent.js:12]: prefer async await over promises

File: src/logger.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+const enumerateErrorFormat = winston.format((info) => {
+  if (info instanceof Error) {
+    Object.assign(info, { message: info.stack });
+  }
+  return info;
+});
+
+const logger = winston.createLogger({
+  level: ENV !== 'prod' ? 'info' : 'debug',
```
Comment (mehul-mittal-credgenics) [inline src/logger.js:12]: hard coded log level

File: src/logger.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  format: winston.format.combine(
+    enumerateErrorFormat(),
+    ENV !== 'prod' ? winston.format.colorize() : winston.format.uncolorize(),
```
Comment (mehul-mittal-credgenics) [inline src/logger.js:15]: remove env based checks from every where

File: src/routes/index.js [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+const router = express.Router();
+
+const defaultRoutes = [
+  {
+    path: '/',
+    route: Route,
+  },
+];
+
+defaultRoutes.forEach((route) => {
```
Comment (mehul-mittal-credgenics) [inline src/routes/index.js:13]: don't need unecessary loop

File: src/routes/route-controller.js [hunk 11]
```diff
@@ -0,0 +1,47 @@
+const httpStatus = require('http-status');
+const catchAsync = require('../utils/catchAsync');
+const {
+  createSchEvent,
```
Comment (mehul-mittal-credgenics) [inline src/routes/route-controller.js:4]: remove abbrevations

File: src/routes/route-controller.js [hunk 12]
```diff
@@ -0,0 +1,47 @@
+const httpStatus = require('http-status');
+const catchAsync = require('../utils/catchAsync');
```
Comment (mehul-mittal-credgenics) [inline src/routes/route-controller.js:2]: fix nomenclature

File: src/routes/route-controller.js [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+const {
+  createSchEvent,
+  fetchSchEvents,
+  deleteSchEvent,
+  updateSchEvent,
+} = require('../services/service');
+const logger = require('../logger');
+
+const createEvent = catchAsync(async (req, res) => {
```
Comment (mehul-mittal-credgenics) [inline src/routes/route-controller.js:11]: catch async can be a middleware

File: src/routes/route-controller.js [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+  const _id = await createSchEvent(req.body);
+  logger.info(`And the logger is ${_id}`);
+  res
```
Comment (mehul-mittal-credgenics) [inline src/routes/route-controller.js:14]: res.json can be used

File: src/routes/route.js [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+  '/update',
+  validate(reqValidation.updateEvent),
+  reqController.updateEvent,
+);
+router.delete(
+  '/delete',
+  validate(reqValidation.deleteEvent),
+  reqController.purgeEvent,
+);
+router.get('/healthz', (req, res) => {
```
Comment (mehul-mittal-credgenics) [inline src/routes/route.js:28]: this route should be with base route

File: src/services/service.js [hunk 16]
```diff
@@ -0,0 +1,147 @@
...
+const mongoose = require('mongoose');
+const ApiError = require('../utils/ApiError');
+const logger = require('../logger');
+const agenda = require('../agendaInit');
+const jobDefination = require('../jobs/jobs_list/triggerEvent');
+const SchedulerModel = require('../mongoModel');
+
+const createSchEvent = async (body) => {
+  try {
+    const { body_ } = body;
```
Comment (mehul-mittal-credgenics) [inline src/services/service.js:12]: can be reduced to lesser line

**Review by mehul-mittal-credgenics — CHANGES_REQUESTED** [review]
Incorporate suggested changes and review the code yourself before submitting

File: README.md [hunk 17]
```diff
@@ -1,2 +1 @@
-# credgenics-remarks-service
 # scheduler-service
```
[the remark this answers is not in the document]
Comment (Bashar200) [reply README.md:1]: done

File: package.json [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+  "main": "index.js",
+  "scripts": {
+    "test": "mocha",
+    "start": "node ./src/app.js",
+    "lint": "eslint .",
+    "lint:fix": "eslint . --fix"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/Bashar200/scheduler-service.git"
```
[the remark this answers is not in the document]
Comment (Bashar200) [reply package.json:14]: done

File: src/agendaInit.js [hunk 19]
```diff
@@ -0,0 +1,36 @@
+const { Agenda } = require('@hokify/agenda');
```
[the remark this answers is not in the document]
Comment (Bashar200) [reply src/agendaInit.js:1]: done

File: src/agendaInit.js [hunk 20]
```diff
@@ -0,0 +1,36 @@
...
+const { MONGO_URI } = require('./settings');
+const logger = require('./logger');
+
+const agenda = new Agenda({
+  db: { address: `${MONGO_URI}/agenda`, collection: 'agendaJobs' },
+});
+agenda.maxConcurrency(10);
```
[the remark this answers is not in the document]
Comment (Bashar200) [reply src/agendaInit.js:8]: done

File: src/agendaInit.js [hunk 21]
```diff
@@ -0,0 +1,36 @@
...
+
+const jobTypes = ['triggerEvent'];
+
+agenda.on('ready', async () => {
+  const jobsAvailable = await agenda.jobs({ type: 'normal' });
+  jobsAvailable.forEach((_) => {
+    jobTypes.forEach((type) => {
+      require(`./jobs/jobs_list/${type}`)(
```
[the remark this answers is not in the document]
Comment (Bashar200) [reply src/agendaInit.js:16]: done

File: src/agendaInit.js (hunk 1 above)
Comment (Bashar200) [reply src/agendaInit.js:29]: done

File: src/routes/index.js (hunk 10 above)
Comment (Bashar200) [reply src/routes/index.js:13]: done

File: src/services/service.js (hunk 16 above)
Comment (Bashar200) [reply src/services/service.js:12]: done

File: src/routes/route-controller.js (hunk 11 above)
Comment (Bashar200) [reply src/routes/route-controller.js:4]: done

File: src/logger.js (hunk 8 above)
Comment (Bashar200) [reply src/logger.js:12]: done

File: src/logger.js (hunk 9 above)
Comment (Bashar200) [reply src/logger.js:15]: done

File: src/routes/route.js (hunk 15 above)
Comment (Bashar200) [reply src/routes/route.js:28]: done

File: src/routes/route-controller.js (hunk 14 above)
Comment (Bashar200) [reply src/routes/route-controller.js:14]: done

File: src/error.js (hunk 3 above)
Comment (Bashar200) [reply src/error.js:9]: done

File: src/error.js (hunk 4 above)
Comment (Bashar200) [reply src/error.js:21]: done

File: src/jobs/jobs_list/triggerEvent.js (hunk 5 above)
Comment (Bashar200) [reply src/jobs/jobs_list/triggerEvent.js:1]: done

File: src/app.js (hunk 2 above)
Comment (Bashar200) [reply src/app.js:17]: done

File: src/jobs/jobs_list/triggerEvent.js (hunk 7 above)
Comment (Bashar200) [reply src/jobs/jobs_list/triggerEvent.js:12]: done

File: src/jobs/jobs_list/triggerEvent.js (hunk 6 above)
Comment (Bashar200) [reply src/jobs/jobs_list/triggerEvent.js:11]: done

File: src/routes/route-controller.js (hunk 13 above)
Comment (Bashar200) [reply src/routes/route-controller.js]: done

File: Dockerfile [hunk 22]
```diff
@@ -0,0 +1,15 @@
+FROM node:16.16.0
```
[the remark this answers is not in the document]
Comment (Bashar200) [reply Dockerfile:1]: created ticket for the same

**Comment by mehul-mittal-credgenics** [comment]
Repeated Issues

**Pull request closed without merging**